datajunction-clients/python/datajunction/_internal.py

```diff
@@ -582,7 +588,7 @@ def _create_tag(
         response = self._session.post(
             "/tags/",
             timeout=self._timeout,
-            json=tag.dict(exclude_none=True),
+            json=tag.to_dict(),
```

Comment:
Is there a reason why we use a `to_dict()` method sometimes and `asdict(...)` sometimes? I believe asdict would also convert any nested dataclasses.

Reply:
No reason other than the PR took me a while and I was going back and forth between solutions ;)

Reply:
Actually there is one more I forgot to mention.... dataclasses asdict() method does not handle nested dataclasses well:
```
!!! Recursion detected (same locals & position)
```